iam-login-service/src/main/java/it/infn/mw/iam/api/tokens/service/DefaultAccessTokenService.java

```diff
@@ -0,0 +1,107 @@
+package it.infn.mw.iam.api.tokens.service;
+
+import it.infn.mw.iam.api.tokens.converter.TokensConverter;
+import it.infn.mw.iam.api.tokens.exception.TokenNotFoundException;
+import it.infn.mw.iam.api.tokens.model.AccessToken;
+import it.infn.mw.iam.api.tokens.model.TokensListResponse;
+import it.infn.mw.iam.api.tokens.service.filtering.TokensFilterRequest;
+import it.infn.mw.iam.api.tokens.service.paging.OffsetPageable;
+import it.infn.mw.iam.api.tokens.service.paging.TokensPageRequest;
+import it.infn.mw.iam.persistence.repository.IamOAuthAccessTokenRepository;
+
+import org.mitre.oauth2.model.OAuth2AccessTokenEntity;
+import org.mitre.oauth2.service.impl.DefaultOAuth2ProviderTokenService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DefaultAccessTokenService implements TokenService<AccessToken> {
+
+  @Autowired
+  private TokensConverter accessTokenConverter;
+
+  @Autowired
+  private DefaultOAuth2ProviderTokenService tokenService;
+
+  @Autowired
+  private IamOAuthAccessTokenRepository tokenRepository;
+
+  @Override
+  public AccessToken getToken(Long id) throws TokenNotFoundException {
+
+    OAuth2AccessTokenEntity at = getAccessTokenById(id).orElseThrow(TokenNotFoundException::new);
```

Comment:
This exception should likely bring information on the id that failed token resolution